chain-impl-mockchain/src/transaction/mod.rs

```diff
@@ -25,8 +25,13 @@ impl<Extra: property::Serialize> property::Serialize for AuthenticatedTransactio
     type Error = Extra::Error;
 
     fn serialize<W: std::io::Write>(&self, mut writer: W) -> Result<(), Extra::Error> {
-        assert_eq!(self.transaction.inputs.len(), self.witnesses.len());
-
+        if self.transaction.inputs.len() != self.witnesses.len() {
```

Comment:
This shouldn't be enforced by the serialization layer, please remove completely

Reply:
I've moved this check to JCLI